index.js

```diff
@@ -56,6 +56,10 @@ function transact (fn, cb) {
 }
 
 function fastifyPostgres (fastify, options, next) {
+  let pg = defaultPg
+  if (options.pg) {
```

Comment:
While this seems legit to be able to pass `pg` into it, but I don't think there's a good use case.

At this line `pg === defaultPg` is actually `true` even if you pass the `pg` instance after patching it with `pg-range`, so not sure if there's any benefit in passing it altogether.

All `pg-range` does is patch the `pg`, it doesn't return a new/different object. So just `require`ing `pg-range` **before** `fastify-postgres` should do the trick.

Or am I missing something?

Reply:
While I am not a continuous postgres user, it looks like `require('pg')` returns a factory and plugins are installed to that factory instance only so you need to pass that factory instance in?

@temsa do you mind trying the suggestion here to see it works?

Reply:
I agree @cemremengu.

Also, the main reason I'm skeptical about this is that `fastify-postgres` is a dedicated library for `pg`. We do not support any other similar postgres module. Which is why I think allowing the user to pass a pg instance is not so intuitive because the user is free to pass any object and we have no control over it.

Reply:
@mcollina what do you think about this?

Reply:
This also enables mocking, which might be a good feature to support.